### PR TITLE
Improve Parametric Scopes documentation [ci-skip]

### PR DESCRIPTION
**Pull request by klevo**
## Motivation / Background

Documentation for Named Parameter in Scopes lacks detailed explanation on how to use this feature in combination with `url_for` and other helpers that depend on `url_for` like `form_with`. People struggle to make this work, as evidenced by [this discussion](https://stackoverflow.com/questions/61236638/how-to-set-scoped-param-value-in-form-with) for example.

## Detail

This Pull Request expands the documentation and provides an example that works in conjunction with `form_with` and other helpers that depend on `url_for`.

## Checklist

Before submitting the PR make sure the following are checked:

- [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
- [x] There are no typos in commit messages and comments.
- [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: [Fix #issue-number]
- [x] Feature branch is up-to-date with main (if not - rebase it).
- [x] Pull request only contains one commit for bug fixes and small features. If it's a larger feature, multiple commits are permitted but must be descriptive.
- [x] Tests are added if you fix a bug or add a feature.
- [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
- [x] PR is not in a draft state.
- [x] CI is passing.

### CODE REVIEW

**Review by jonathanhefner — COMMENTED** [review]
I think this would be a helpful addition! :+1:

**Comment by klevo** [comment]
Thanks for the review @jonathanhefner - I learned a lot from it. All suggestions implemented and I rebased the branch into a single commit.

File: guides/source/routing.md [hunk 1]
```diff
@@ -1164,15 +1164,29 @@ resources to use `photos_path` and `accounts_path`.
...
+
+```ruby
+account_article_path(@account, @article) # => /1/article/9
+url_for([@account, @article]) # => /1/article/9
+form_with(model: [@account, @article]) # => <form action="/1/article/9" ...>
+```
+
+We [use a constraint](#segment-constraints) to limit the scope to only match ID-like strings.
+
+NOTE: The `:as` and `:constraints` parameters are optional, however without the use of `:as` such approach will result in an error when combined with `url_for` helper directly or methods like `form_with` that use `url_for` internally. For example, Rails won't be able to build a route from `url_for([@account, @article])` unless you specified `:as`.
```
Comment (jonathanhefner) [inline guides/source/routing.md:1189]: Nitpick: in the guides, there is a tendency to overuse callouts like `NOTE:`.  When possible, I think it is better to integrate the text with the surrounding content, to stave off callout fatigue.

One way to do that here might be:

```suggestion
We are [using a constraint](#segment-constraints) to limit the scope to only match ID-like strings. You can change the constraint to suit your needs, or omit it entirely. The `:as` option is also not strictly required, but without it, Rails will raise an error when evaluating `url_for([@account, @article])` or other helpers that rely on `url_for`, such as `form_with`.
```

**Comment by klevo** [comment]
@jonathanhefner Perfect, that indeed reads so much better. 

I have also aligned the `# =>` in the code sample vertically, as per documentation guidelines, and linked the `url_for` and `form_with` to respective `api.rubyonrails.org` documentation, as I noticed in other sections of the guides.

**Comment by jonathanhefner** [comment]
To explain a bit about linking to the API docs: what we usually do is link the first occurrence of a method name in a guide, similar to how Wikipedia links the first occurrence of term in a wiki article.  Sometimes it makes sense to link additional occurrences; sometimes it makes sense to not link any occurrence, because the method isn't the focus of the guide.

Here, I think it's fine to link `url_for`, but it would be good to link the first occurrence under https://guides.rubyonrails.org/v7.0.4/routing.html#creating-paths-and-urls-from-objects instead.  I'm on the fence about `form_with`, since it's not really related to routing, but it won't hurt.

Also, I feel like reference-style links (i.e. `[][]` syntax, instead of `[]()` syntax) help keep the document readable when editing.  You can also re-use them if you do link additional occurrences of a method name.

I went ahead and pushed a commit with these minor changes, but otherwise everything looks great!  Thank you, @klevo! :tada:

**Comment by klevo** [comment]
Many thanks @jonathanhefner 🤝

**Pull request merged**